### ISSUE TITLE
A variant of \ifsingletitle

### ISSUE THREAD

**Issue by maieul** [plk/biber#121]
Considering this MWE:

`````
\documentclass{article}
\usepackage[singletitle,labeltitle=true]{biblatex}
\usepackage{filecontents}

\DeclareLabeltitle{%
  \field{shorttitle}
  \field{title}
  \field{maintitle}
}

\begin{filecontents}{\jobname.bib}

@book{Rouquette2012,
	Author={Maïeul Rouquette},
	Title={(Xe)LaTeX appliqué aux sciences humaines}
}
@mvbook{Ehrhard,
	Address = {Osnabrück},
	Author = {Albert Ehrhard},
	Origdate = {1937/1952},
	Origlocation = {Leipzig},
	Origpublisher = {J. C. Hinrichs Verlag},
	Publisher = {Otto Zeller Verlag},
	Subtitle = {von den Anfängen bis zum Ende des 16. Jahrhunderts},
	Title = {Überlieferung und Bestand der hagiographischen und homiletischen Literatur der griechischen Kirche},
	Volumes = {3},
	Year = {1965}}

@book{Ehrhard1.1,
	Crossref = {Ehrhard},
	Origdate = {1937},
	Part = {1},
	Volume = {1}
}

@book{Ehrhard1.2,
	Crossref = {Ehrhard},
	Origdate = {1938},
	Part = {2},
	Volume = {1},
}

@book{Ehrhard1.3,
	Crossref = {Ehrhard},
	Date-Modified = {2014-06-03 12:48:11 +0000},
	Keywords = {a verifier,prague_secondaire},
	Origdate = {1939/1952},
	Origlocation = {Leipzig},
	Origpublisher = {J. C. Hinrichs Verlag},
	Part = {3},
	Publisher = {Otto Zeller Verlag},
	Volume = {1},
	Year = {1965}
}
\end{filecontents}
\bibliography{\jobname.bib}
\begin{document}
\nocite{*}
s
\end{document}
````

Conforming to the biblatex handbook, the `\ifsingletitle` is set to true for `Rouquette2012` but not for the other one. 

However, it could be useful to have a variant (with which name ?) which will be set to TRUE if a combination of labelname + titlename is unique. Will useful to print something like  `Ehrahrd, op. cit, t. 1.3, xxx`

**Comment by plk** [comment]
Do you mean that `singletitle` should be be true for the entries which inherit the `mvbook` because technically it is a single title, just shared with other volumes?

**Comment by maieul** [comment]
yes, but as am afraid to break ascendant compatibility, I was asking for a new test…

**Comment by plk** [comment]
Hmm, I need to think about this - I think this is potentially a bug.

**Comment by maieul** [comment]
Technically, that is on the letter of hanbook. But maybe not on the spirit :-)

**Comment by maieul** [comment]
ps : in general there is some limit with the biblatex op cit style when using multiple volument book with shared title. I do some test on my computer and maybe I will release a fix package �

**Comment by plk** [comment]
Shall I wait until you have done this before I look into this? I think it's quite a hard problem.

**Comment by maieul** [comment]
With my test, only in a little phase, all is ok except this case. But it is complex for me to split between my internal hack and a generalist package. So for now, I have no idea in which time I could send a "bugfix" package.

**Comment by maieul** [comment]
some feedback on this issue?

**Comment by plk** [comment]
Does the enhancement in plk/biblatex#438 help with this?

**Comment by maieul** [comment]
no, the problem plk/biblatex#438 is independant. It concern the hash, while this problem concern the `\ifsingletitle` tool. I have tested with biber 2.6, and the `singletitle` field is still true only for Rouquette2012 in this example, and not for Ehrhard  volume

**Comment by plk** [comment]
Please try latest 3.5/2.6. I have implemented the option `uniquework` which enables the test `\ifuniquework` which is only true when `labelname+labeltitle` is unique. So the possibilities are now:

`\ifsingletitle` - true when labelname (if no labelname, labeltitle) is unique
`\ifuniquetitle` - true when labeltitle is unique
`\ifuniquework` - true when labelname+labeltitle is unique

**Comment by maieul** [comment]
Are your sure to have send the good version on sourceforge? With these files, `uniquework` is not inserted in the .bbl file, and LaTeX tell me that `\ifuniquework` is not defined.

**Comment by maieul** [comment]
my bad: `\ifuniquework` is defined, but it seems that biber don't add the uniquework data in the .bbl (I used the MWE below)

**Comment by plk** [comment]
Hmm, is uniquework set to 1 in the .bcf?

**Comment by maieul** [comment]
it was not. But I have added the uniquework option when loading biblatex. It set uniquework to 1 in the .bcf file, but that has not changed the .bbl file

**Comment by plk** [comment]
Ok, there are two aspects to this, there is the new test and then there is the inheritance of it ... looking at it.

**Comment by plk** [comment]
This is a hard problem in fact. Inheritance has already happened by the time uniquework tracking information is generated. Hmm.

**Comment by maieul** [comment]
�

**Comment by maieul** [comment]
Hum, indeed in my case the problem is not to test the unicity of labelname+labeltitle (which can't be unic…) but to consider that the `singletitle` test (or variant) should be made considering inherited labelname as a unic labelname.

**Comment by plk** [comment]
This is very difficult, if not impossible to manage in biber because the data which these tests need have to be generated after inheritance is already done. However, I think you could do this in your style/code by using `\entrydata` via the contents of the `crossref` field (which still exists in the .bbl) and you can then get the value of `singletitle` etc. for the inheritance children.

**Comment by maieul** [comment]
> and you can then get the value of singletitle etc. for the inheritance children.

that is the same problem, isn't it ? singletitle won't be set to true after inheritance (cf the result with this MWE). 

What I imagine in biber:
- Inheritance process. 
- checking ifsingletitle and other thing. Before doing it, delete in the list of entry to test the entry which labeltitle / authortitle field are inherited from other entry.
- reporting ifsingletitle and other thing in the entry deleted in the previous step.

I know it is more easy to write that to code, but�

**Comment by plk** [comment]
Believe me, I've looked at this several ways in biber and it is basically impossible without messy hacks. It can't be done by normal inheritance because these are not data fields and should only be inherited for specific relationships (mvbook etc.) I think this is better done in biblatex with `\entrydata`.

**Comment by maieul** [comment]
I believe you� but I don't see how I could use entrydata to get a ifsingletitle information that won't be here.

**Comment by maieul** [comment]
may I propose you that: don't take time on this point, but let the issue open. I will look myself on the biber code, and maybe I will find an solution

**Comment by plk** [comment]
In you test for `singletitle`, you do a test for `crossref` first. If that exists, use `\entrydata` to look in the crossrefed entry and check if `singletitle` is set there and if it is, do the same as if `\ifsingletitle` had been true.

**Comment by maieul** [comment]
The problem is that the singletitle is not set to true even for the main entry.

Please look on the bbl result of the previous MWE.

```
\refsection{0}
  \sortlist[entry]{nty/global/}
    \entry{Ehrhard1.1}{book}{}
      \name{author}{1}{}{%
        {{hash=a22f734befe97f5497a03a75be4da471}{%
           family={Ehrhard},
           family_i={E\bibinitperiod},
           given={Albert},
           given_i={A\bibinitperiod}}}%
      }
      \list{location}{1}{%
        {Osnabrück}%
      }
      \list{origlocation}{1}{%
        {Leipzig}%
      }
      \list{origpublisher}{1}{%
        {J. C. Hinrichs Verlag}%
      }
      \list{publisher}{1}{%
        {Otto Zeller Verlag}%
      }
      \strng{namehash}{a22f734befe97f5497a03a75be4da471}
      \strng{fullhash}{a22f734befe97f5497a03a75be4da471}
      \strng{authornamehash}{a22f734befe97f5497a03a75be4da471}
      \strng{authorfullhash}{a22f734befe97f5497a03a75be4da471}
      \field{sortinit}{E}
      \field{sortinithash}{fefc5210ef4721525b2a478df41efcd4}
      \field{extratitle}{1}
      \field{labelnamesource}{author}
      \field{labeltitlesource}{maintitle}
      \strng{crossref}{Ehrhard}
      \field{mainsubtitle}{von den Anfängen bis zum Ende des 16. Jahrhunderts}
      \field{maintitle}{Überlieferung und Bestand der hagiographischen und homiletischen Literatur der griechischen Kirche}
      \field{origyear}{1937}
      \field{part}{1}
      \field{volume}{1}
      \field{volumes}{3}
      \field{year}{1965}
    \endentry
    \entry{Ehrhard1.2}{book}{}
      \name{author}{1}{}{%
        {{hash=a22f734befe97f5497a03a75be4da471}{%
           family={Ehrhard},
           family_i={E\bibinitperiod},
           given={Albert},
           given_i={A\bibinitperiod}}}%
      }
      \list{location}{1}{%
        {Osnabrück}%
      }
      \list{origlocation}{1}{%
        {Leipzig}%
      }
      \list{origpublisher}{1}{%
        {J. C. Hinrichs Verlag}%
      }
      \list{publisher}{1}{%
        {Otto Zeller Verlag}%
      }
      \strng{namehash}{a22f734befe97f5497a03a75be4da471}
      \strng{fullhash}{a22f734befe97f5497a03a75be4da471}
      \strng{authornamehash}{a22f734befe97f5497a03a75be4da471}
      \strng{authorfullhash}{a22f734befe97f5497a03a75be4da471}
      \field{sortinit}{E}
      \field{sortinithash}{fefc5210ef4721525b2a478df41efcd4}
      \field{extratitle}{2}
      \field{labelnamesource}{author}
      \field{labeltitlesource}{maintitle}
      \strng{crossref}{Ehrhard}
      \field{mainsubtitle}{von den Anfängen bis zum Ende des 16. Jahrhunderts}
      \field{maintitle}{Überlieferung und Bestand der hagiographischen und homiletischen Literatur der griechischen Kirche}
      \field{origyear}{1938}
      \field{part}{2}
      \field{volume}{1}
      \field{volumes}{3}
      \field{year}{1965}
    \endentry
    \entry{Ehrhard1.3}{book}{}
      \name{author}{1}{}{%
        {{hash=a22f734befe97f5497a03a75be4da471}{%
           family={Ehrhard},
           family_i={E\bibinitperiod},
           given={Albert},
           given_i={A\bibinitperiod}}}%
      }
      \list{location}{1}{%
        {Osnabrück}%
      }
      \list{origlocation}{1}{%
        {Leipzig}%
      }
      \list{origpublisher}{1}{%
        {J. C. Hinrichs Verlag}%
      }
      \list{publisher}{1}{%
        {Otto Zeller Verlag}%
      }
      \strng{namehash}{a22f734befe97f5497a03a75be4da471}
      \strng{fullhash}{a22f734befe97f5497a03a75be4da471}
      \strng{authornamehash}{a22f734befe97f5497a03a75be4da471}
      \strng{authorfullhash}{a22f734befe97f5497a03a75be4da471}
      \field{sortinit}{E}
      \field{sortinithash}{fefc5210ef4721525b2a478df41efcd4}
      \field{extratitle}{3}
      \field{labelnamesource}{author}
      \field{labeltitlesource}{maintitle}
      \strng{crossref}{Ehrhard}
      \field{mainsubtitle}{von den Anfängen bis zum Ende des 16. Jahrhunderts}
      \field{maintitle}{Überlieferung und Bestand der hagiographischen und homiletischen Literatur der griechischen Kirche}
      \field{origendyear}{1952}
      \field{origyear}{1939}
      \field{part}{3}
      \field{volume}{1}
      \field{volumes}{3}
      \field{year}{1965}
      \keyw{a verifier,prague_secondaire}
    \endentry
    \entry{Ehrhard}{mvbook}{}
      \name{author}{1}{}{%
        {{hash=a22f734befe97f5497a03a75be4da471}{%
           family={Ehrhard},
           family_i={E\bibinitperiod},
           given={Albert},
           given_i={A\bibinitperiod}}}%
      }
      \list{location}{1}{%
        {Osnabrück}%
      }
      \list{origlocation}{1}{%
        {Leipzig}%
      }
      \list{origpublisher}{1}{%
        {J. C. Hinrichs Verlag}%
      }
      \list{publisher}{1}{%
        {Otto Zeller Verlag}%
      }
      \strng{namehash}{a22f734befe97f5497a03a75be4da471}
      \strng{fullhash}{a22f734befe97f5497a03a75be4da471}
      \strng{authornamehash}{a22f734befe97f5497a03a75be4da471}
      \strng{authorfullhash}{a22f734befe97f5497a03a75be4da471}
      \field{sortinit}{E}
      \field{sortinithash}{fefc5210ef4721525b2a478df41efcd4}
      \field{extratitle}{4}
      \field{labelnamesource}{author}
      \field{labeltitlesource}{title}
      \field{origendyear}{1952}
      \field{origyear}{1937}
      \field{subtitle}{von den Anfängen bis zum Ende des 16. Jahrhunderts}
      \field{title}{Überlieferung und Bestand der hagiographischen und homiletischen Literatur der griechischen Kirche}
      \field{volumes}{3}
      \field{year}{1965}
    \endentry
    \entry{Rouquette2012}{book}{}
      \name{author}{1}{}{%
        {{hash=9828ee193af02294d2bca32cb8321c67}{%
           family={Rouquette},
           family_i={R\bibinitperiod},
           given={Maïeul},
           given_i={M\bibinitperiod}}}%
      }
      \strng{namehash}{9828ee193af02294d2bca32cb8321c67}
      \strng{fullhash}{9828ee193af02294d2bca32cb8321c67}
      \strng{authornamehash}{9828ee193af02294d2bca32cb8321c67}
      \strng{authorfullhash}{9828ee193af02294d2bca32cb8321c67}
      \field{sortinit}{R}
      \field{sortinithash}{c7387613477035a752d935acfc3e3ea2}
      \true{singletitle}
      \field{labelnamesource}{author}
      \field{labeltitlesource}{title}
      \field{title}{(Xe)LaTeX appliqué aux sciences humaines}
    \endentry
  \endsortlist
\endrefsection

```

The only entry with a singletitle is "Rouquette2012", while "Ehrhard" as no singletitle.

**Comment by plk** [comment]
Oh, sorry, I see. Let me think about it more. I don't think it can be dealt with via inheritance.

**Comment by maieul** [comment]
maybe a solution: allow to tell in the bib file that a entry should not be included in the test for ifsingletitle? but if biber is able of that� I don't see why it could not be able to automatically exclude these �

**Comment by plk** [comment]
I think to be really flexible, it will need a new biblatex option to specific which child/parent inheritance pairs should have the same `singletitle`, `uniquetitle` and `uniquework` settings. This won't be desirable for all inheritance pairs. Can you think of a default list of those which would?

**Comment by maieul** [comment]
hum. in my aim, that is conditionned by the relation between the labeltitle of the child and the labeltitle of the the entry. If equal> don't be take.

For example, in this MWE, we have `@mvbook->@book` relationship. All the individual volume have the same title that the ensemble. But in some other case, for `@mvbook->@book` we could have individual volume with individual title.

**Comment by plk** [comment]
Then the `ifuniquework` test is better for this?

**Comment by maieul** [comment]
Sincerily, I don't understand in which case `\ifsinglework` could be useful. 

Suppose, for example, that I have

```
@book{A1
	Author={A},
	Title={1}
}

@book{A2
	Author={A},
	Title={2}
}
````

In this case, for both entry I will I have uniquework set to TRUE. But� what does matter? It is just a tautology. 


In the previous MWE, the ideal would be that {Ehrhard+Ehrhard1.1+Ehrhard1.2+Ehrhard1.3 would be tested as one block when biber set the ifsingletitle in the .bbl. "One block" could be defined by the conjunction of three critera:
- crossref relationship (potentially multiple and / or nested)
- equality between labeltitle field of all the entry of the block
- equality between labelname field of all entry of the block

**Comment by plk** [comment]
When `uniquework` is false, both `uniquetitle` and `singletitle` would be false but not the converse.

```
% singletitle=false
% uniquetitle=false
% uniquework=true
@book{test1,
  AUTHOR = {Author1},
  TITLE  = {Title1}
}

% singletitle=false
% uniquetitle=true
% uniquework=true
@book{test2,
  AUTHOR = {Author1},
  TITLE  = {Title2}
}

% singletitle=false
% uniquetitle=false
% uniquework=false
@book{test3,
  AUTHOR = {Author2},
  TITLE  = {Title1},
  VOLUME = {1}
}

% singletitle=false
% uniquetitle=false
% uniquework=false
@book{test4,
  AUTHOR = {Author2},
  TITLE  = {Title1},
  VOLUME = {2}
}

```

**Comment by maieul** [comment]
what I don't understand is: in which case it would be useful to test uniquework?

**Comment by maieul** [comment]
oh, your mean, that, in my case all the Ehrhard book would be uniquework=false. 

So if I need an "op. cit." for these book, I should test:
- First uniquework
- If true, so I test singletitle
- If false, what should I test ? 

IMHO, the uniquework is not relevant.

**Comment by plk** [comment]
I don't think it's relevant to your case but people sometimes want to do things like archiving work where a method for detecting duplicate records (with different citekeys) might help. This would be a reasonable test. Perhaps multiple people add duplicate .bib records to the same database with different keys etc. But, I think it's not helpful to you here, I have to think about something different for your issue.

**Comment by maieul** [comment]
Oh, yes. In this case it could be useful. Maybe you could add this example in the handbook ?

**Comment by plk** [comment]
I have an idea how to do this using a new option in the inheritance specification allowing a particular inheritance to not contribute to the singletitle tracking. I will try to get a test implementation tomorrow for you.

**Comment by maieul** [comment]
many thank

**Comment by plk** [comment]
I have a draft working now with 3.5/2.6 DEV versions. Any of these 3 inheritance specifications make your example work for me now:

```
\DefaultInheritance{all=true,override=false,suppress=singletitle}

\DefaultInheritance[
  \except{mvbook}{book}{suppress=singletitle}
]{all=true,override=false}

\DeclareDataInheritance[suppress=singletitle]{mvbook}{book}{%
  \inherit{author}{author}
  \inherit{author}{bookauthor}
}
```

Note - you only need one of them - they are just for demonstration of the different scopes at which you can apply this. The documentation is updated and this suppression works for `singletitle`, `uniquetitle` and `uniquework` (or several at once).

One question - at the moment it means that `singletitle` is true in `Ehrhard`, `Ehrhard1.1`, `Ehrhard1.2` and `Ehrhard1.3`. It is easy to make it only true in `Ehrhard`.  Which would you expect? I think I would expect the current behaviour as the suppression only suppresses the *tracking* of the singletitle etc. information. Let me know what you think.

**Comment by maieul** [comment]
Many thanks. I will try all these thing next week, if possible.

**Comment by maieul** [comment]
Hi,

I have found time to make test.

Here my feedback. 

My MWE is the following:

```
\documentclass{article}
\usepackage[singletitle,labeltitle=true]{biblatex}
\usepackage{filecontents}

\DeclareDataInheritance[suppress=singletitle]{mvbook}{book}{%
  \inherit{author}{author}
  \inherit{author}{bookauthor}
  \inherit{title}{title}
  \inherit{title}{booktitle}
}

\begin{filecontents}{\jobname.bib}


@mvbook{Zacos1972,
	Address = {Bâle},
	Author = {Zacos, George and Veglery, Alexander},
	Publisher = {J. J. Augustin},
	Title = {Byzantine Lead Seals},
	Volumes = {2},
	Year = {1972}}

@book{Zacos1972.1.1,
	Crossref = {Zacos1972},
	Part = {1},
	Title = {Nos. 1-1095, Imperial seals, V\textsuperscript{th} to XV\textsuperscript{th} centuries ; non-imperial seals, VI\textsuperscript{th} to IX\textsuperscript{th} centuries},
	Volume = {1}}

@book{Zacos1972.1.2,
	Crossref = {Zacos1972},
	Part = {2},
	Title = {Nos. 1096-2671A, non-imperial seals, VI\textsuperscript{th} to IX\textsuperscript{th} centuries},
	Volume = {1}}

@book{Zacos1972.1.3,
	Crossref = {Zacos1972},
	Part = {3},
	Title = {Nos. 2672-3231, imperial and allied seals, V\textsuperscript{th} to XIV\textsuperscript{th} centuries ; non-imperial seals, VI\textsuperscript{th} to IX\textsuperscript{th} centuries},
	Volume = {1}}


@book{Rouquette2012,
    Author={Maïeul Rouquette},
    Title={(Xe)LaTeX appliqué aux sciences humaines}
}
@mvbook{Ehrhard,
    Address = {Osnabrück},
    Author = {Albert Ehrhard},
    Origdate = {1937/1952},
    Origlocation = {Leipzig},
    Origpublisher = {J. C. Hinrichs Verlag},
    Publisher = {Otto Zeller Verlag},
    Subtitle = {von den Anfängen bis zum Ende des 16. Jahrhunderts},
    Title = {Überlieferung und Bestand der hagiographischen und homiletischen Literatur der griechischen Kirche},
    Volumes = {3},
    Year = {1965}}

@book{Ehrhard1.1,
    Crossref = {Ehrhard},
    Origdate = {1937},
    Part = {1},
    Volume = {1}
}

@book{Ehrhard1.2,
    Crossref = {Ehrhard},
    Origdate = {1938},
    Part = {2},
    Volume = {1},
}

@book{Ehrhard1.3,
    Crossref = {Ehrhard},
    Date-Modified = {2014-06-03 12:48:11 +0000},
    Keywords = {a verifier,prague_secondaire},
    Origdate = {1939/1952},
    Origlocation = {Leipzig},
    Origpublisher = {J. C. Hinrichs Verlag},
    Part = {3},
    Publisher = {Otto Zeller Verlag},
    Volume = {1},
    Year = {1965}
}
\end{filecontents}
\bibliography{\jobname.bib}
\begin{document}
\nocite{*}
s
\end{document}
```

1) singletitle is true in Ehrhard, Ehrhard1.1, Ehrhard1.2 and Ehrhard1.3, that is what I expected. Many thank.
2) But, as explained, a entrytype setting is not the more relevant. Setting using labeltitle/labelfield are more relevant. See the case of the "Zacos" book above. In this case, all individidual @book of the global @mvbook have its own title > singletitle should be set to false.
3) on the handbook "which would trigger the tracking (table 6)is inherited" > lacking space after the parenthesis.

**Comment by plk** [comment]
But in that case, uniquework would be true. Singletitle can't be true for that case because there really is more than one title for the name.

**Comment by maieul** [comment]
yes, and I think it will be expected

**Comment by plk** [comment]
Yes, but that's how you can tell the cases apart in tests - in the Ehrhard entries, only `singletitle` is true but not `uniquework`. The Zacos texts have both `singletitle` and `uniquework`. `Uniquework` is only useful in combination with the other tests in such cases.

**Comment by maieul** [comment]
sorry, I don't understand your answer (my bad english)

could you reformulate it ?

**Comment by plk** [comment]
Apologies, you were right, the logic in biber needed to be changed - please get biber 2.6 again (and biblatex 3.5). The same example should now work correctly.

**Comment by maieul** [comment]
that is working !

**Comment by plk** [comment]
All fixed now except `\DefaultInheritance{suppress={singletitle,uniquetitle}}` as this is correct I think. Since this is not the biblatex default setting (`all=true` is not specified), titles are not inherited and so it is not expected to work ...

Also, "suppress" is renamed to "ignore" as suggested - this is a better name, I agree.
I have also stopped `singletitle` using labeltitle as a fallback since this is no longer necessary as there is `uniquetitle` now and was always a bit confusing.

You will need to pull 3.5/2.6 again to get the changes.

**Comment by maieul** [comment]
> Since this is not the biblatex default setting (all=true is not specified), titles are not inherited and so it is not expected to work ...

it seems to me that the title are inherited by default, as any field (p. 189 of the actual handbook)



> I have also stopped singletitle using labeltitle as a fallback since this is no longer necessary as there is uniquetitle now and was always a bit confusing.

so "singletitle" is no more available ?

**Comment by plk** [comment]
Yes, titles are inherited by default but only because of the default setting:

`\DefaultInheritance{all=true,override=false}`

If you change this to

`\DefaultInheritance{suppress={singletitle,uniquetitle}}`

there is no more default inheritanc of *any* field without the `all=true`. If a user sets the default like this, it overrides the biblatex default - they are not merged.

Singletitle is still available, it is just that internally, if there was no labelname, it would use labeltitle instead as a fallback which was confusing and could lead to strange results. This doesn't impact your examples.

**Comment by maieul** [comment]
I will do test and make feedback


> Yes, titles are inherited by default but only because of the default setting:
> 
> \DefaultInheritance{all=true,override=false}
> 
> If you change this to
> 
> \DefaultInheritance{suppress={singletitle,uniquetitle}}

you are right

> 
> there is no more default inheritanc of any field without the all=true. If a user sets the default like this, it overrides the biblatex default - they are not merged.
> 
> Singletitle is still available, it is just that internally, if there was no labelname, it would use labeltitle instead as a fallback which was confusing and could lead to strange results. This doesn't impact your examples.

Nice ! Indeed I never used `\ifsingletitle` allone, but `\ifboolexpr{test{\ifsingletitle} and not test {\ifnameundef{labelname}}}`�

**Comment by maieul** [comment]
I have made test.

I come back on `\DefaultInheritance{ignore={singletitle,uniquetitle}}`. Now, singletitle is set to true for the mainentry, despite it should be false. as if `\DefaultInheritance{ignore={singletitle,uniquetitle}}` would be absent.

**Comment by maieul** [comment]
ps: you have not yet update the handbook, which style mention supprress.

**Comment by plk** [comment]
Handbook is updated. The behaviour for `\DefaultInheritance{ignore={singletitle,uniquetitle}}` is correct I think because for the main entries labelname/labeltitle are not inherited and so the `ignore` is irrelevant for them. The `ignore` only effects inherited information.

**Comment by maieul** [comment]
Ok. But singletitletitle is based on labelname, not labeltitle.

Other point.
- If I don't use DefaultInheritance, author field is inherited
- If I use DefautInheritance with `all` option,  author field is inherited
-  If I use DefautInheritance without `all` option,  author field is NOT inherited despite the handbook told use that "all" is set to TRUE by default (p. 189 of the new handbook)

**Comment by plk** [comment]
Singletitle tracks labelname but *ignoring* of singletitle also looks at labeltitle. The othe points are due to the default inheritance rules I think - there is a lot more happening than just the `\DefaultInheritance` rule - see Appendix B of the manual.

**Comment by maieul** [comment]
my point was that the handbook told use that using `DefautInheritance` have the option `all` set to TRUE by default, but, in reality, it is set to false. So with the correction of the handbook. All will be ok.

**Comment by plk** [comment]
Well it is true by default, that is, the default setting is `\DefaultInheritance{all=true,override=false}` in `biblatex.def`. If you change the default, it's no longer the default ...

**Comment by maieul** [comment]
Oh, I see, I was thinking that "default" in the handbook was "default value passed to `DefaultInheritance`" and not "value passed to the default \DefaultInheritance`. My bad. So we can close this issue. Thanks you very, very much for your involving!
